pallets/nodle-staking/src/lib.rs

```diff
@@ -0,0 +1,2311 @@
+        }
+        // Returns Some(Some(balance)) if successful
+        // None if nomination not found
+        // Some(None) if underflow
+        pub fn dec_nomination(
+            &mut self,
+            validator: AccountId,
+            less: Balance,
+        ) -> Option<Option<Balance>> {
+            for x in &mut self.nominations.0 {
+                if x.owner == validator {
+                    if x.amount > less {
+                        x.amount -= less;
+                        // self.total -= less;
+                        self.active_bond -= less;
+                        return Some(Some(x.amount));
+                    } else {
+                        // underflow error; should rm entire nomination if x.amount == validator
+                        return Some(None);
+                    }
+                }
+            }
+            None
+        }
+    }
+
+    impl<AccountId, Balance> Nominator<AccountId, Balance>
+    where
+        AccountId: Ord + Clone,
+        Balance: AtLeast32BitUnsigned + Saturating + Copy,
+    {
+        /// Slash the validator for a given amount of balance. This can grow the value
+        /// of the slash in the case that the validator has less than `minimum_balance`
+        /// active funds. Returns the amount of funds actually slashed.
+        ///
+        /// Slashes from `active` funds first, and then `unlocking`, starting with the
+        /// chunks that are closest to unlocking.
+        pub(crate) fn slash_nomination(
+            &mut self,
+            validator: AccountId,
+            mut value: Balance,
+            minimum_balance: Balance,
+        ) -> Balance {
+            let pre_total = self.total;
+            let total = &mut self.total;
+            let pre_active_bond = self.active_bond;
+            let active_bond = &mut self.active_bond;
+
+            let slash_out_of =
+                |total_remaining: &mut Balance, target: &mut Balance, value: &mut Balance| {
+                    let mut slash_from_target = (*value).min(*target);
+
+                    if !slash_from_target.is_zero() {
+                        *target -= slash_from_target;
+                        *total_remaining = total_remaining.saturating_sub(slash_from_target);
+
+                        if *total_remaining <= minimum_balance {
+                            let diff_val = minimum_balance.saturating_sub(*total_remaining);
+                            *target += diff_val;
+                            *total_remaining += diff_val;
+                            slash_from_target -= diff_val;
+                        }
+                        *value -= slash_from_target;
+                    }
+                };
+
+            for x in &mut self.nominations.0 {
+                if x.owner == validator {
+                    slash_out_of(active_bond, &mut x.amount, &mut value);
+                }
+            }
+
+            *total -= pre_active_bond.saturating_sub(*active_bond);
```

Comment:
prob. want to saturate on `*total` or use one more checked math op

Reply:
:+1:  Moved to saturating_sub()